src/strand-guide/strand-guide.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function(scope) {
+
+	scope.Guide = Polymer({
+		is: 'strand-guide',
+
+		behaviors:[
+			StrandTraits.Falsifiable,
+			StrandTraits.Resolvable,
+			StrandTraits.Stackable,
+			StrandTraits.Refable,
+			StrandTraits.DomGettable
+		],
+
+		properties: {
+			contentTarget: {
+				type: Object,
+				value: function() { return this.$.guideContent; }
+			},
+			useLocalStorage: {
+				type: Boolean,
+				value: false
+			},
+			name: {
+				type: String
+			},
+			suppressGuide: {
+				type: Boolean
+			},
+			scope: {
+				type: Object,
+				notify: true,
+				value: function() { return document; }
+			},
+			stackType:{
+				value: "modal"
+			},
+			hidden: {
+				type:Boolean,
+				value:true,
+				reflectToAttribute:true
+			},
+			showFocus: {
+				type: Boolean,
+				value: true
+			},
+			autoDismiss: {
+				type: Boolean,
+				value: false
+			},
+			noscroll: {
+				type:Boolean,
+				value:false
+			},
+			closeBtn: {
+				type: Boolean,
+				value: true
+			},
+			_currentStep: {
+				type: Number,
+				notify: true
+			},
+			_tooltipData: {
+				type: Array,
+				notify: true
+			},
+			data: {
+				type: Array,
+				notify: true,
+				observer: '_dataChanged'
+			},
+		},
+
+		_previousBodyOverflow: null,
+		_isAttached: false,
+
+		domObjectChanged: function(domObject) {
+			if (!this.data && !this.suppressGuide) {
+				this.data = domObject['guide'];
+			}
+		},
+
+		attached: function() {
+			// Attempt to ensure that the DOM has settled before doing any layout
+			this.async(function() {
+				if (this.data) this._init();
+				this._isAttached = true;
+			});
+		},
+
+		_dataChanged: function(newVal, oldVal) {
+			if (!this.suppressGuide) {
+				// Collect object references for all of the targets
+				newVal.forEach(function(item) {
+					var target = Polymer.dom(this.scope).querySelector('#' + item.target);
+					item.targetRef = target;
+				});
+
+				if (this._isAttached) this._init();
+			}
+		},
+
+		_init: function() {
+			// Trigger tooltip and canvas setup
+			this._tooltipData = this.data;
+			this._currentStep = 0;
+		},
+
+		show: function() {
+			if (!this.suppressGuide) {
+				this.hidden = false;
+				this.$.tooltip.open();
+				if (this.showFocus) this.$$('#focus').updateCanvas();
+
+				// strand-modal scroll suppression
+				if(this.noscroll) {
```

Comment:
careful here, `show()` is a public function, and it looks like repeated calls to `show()` without a `hide()` inbetween could cause undesirable results when `hide()` is called sometime later.  Probably add more conditions beyond just `this.noscroll`